website/src/auth.tsx

```diff
@@ -2,52 +2,71 @@ import { HubCallback } from "@aws-amplify/core"
 import Amplify, { Auth, Hub } from "aws-amplify"
 import React, { createContext, useContext, useState } from "react"
 
-const UserContext = createContext<CognitoUser | null>(null)
+import {
+  CognitoUser,
+} from 'amazon-cognito-identity-js'
+
+type UserContextType = {
+  user: CognitoUser | null;
+  setUser: (value: CognitoUser | null) => void;
+};
+
+export const UserContext = createContext<UserContextType>({} as UserContextType);
 
 export const UserProvider = (props: { children: any }) => {
-  const [user, setUser] = useState<CognitoUser | null>(null)
-  React.useEffect(() => {
-    Amplify.configure({
-      Auth: {
-        // TODO How to get these variables from the server?
-        region: process.env["DAILP_AWS_REGION"],
-        userPoolId: process.env["DAILP_USER_POOL"],
-        userPoolWebClientId: process.env["DAILP_USER_POOL_CLIENT"],
-      },
-    })
-
-    Auth.currentUserPoolUser()
-      .then((user) => setUser(user))
-      .catch(() => setUser(null))
-
-    const listener: HubCallback = async (data) => {
-      switch (data.payload.event) {
-        case "signIn":
-        case "signUp":
-          setUser(await Auth.currentUserPoolUser())
-          break
-        case "signOut":
-          setUser(null)
-          break
-      }
-    }
-    Hub.listen("auth", listener)
-    return () => Hub.remove("auth", listener)
-  }, [])
+  const [user, setUser] = useState<CognitoUser | null>(null);
+
+  // React.useEffect(() => {
```

Comment:
We definitely need something here to restore remembered logins, the cognito library docs have an example of it (use case 16).

Reply:
I will be taking a look at the docs for this and let you know if I have questions.